pkg/channel/multichannelfanout/multi_channel_fanout_event_handler_test.go

```diff
@@ -312,7 +311,7 @@ func TestServeHTTPMessageHandler(t *testing.T) {
 
 			logger := zaptest.NewLogger(t, zaptest.WrapOptions(zap.AddCaller()))
 			reporter := channel.NewStatsReporter("testcontainer", "testpod")
-			h, err := NewMessageHandlerWithConfig(context.TODO(), logger, channel.NewMessageDispatcher(logger), tc.config, reporter, tc.recvOptions...)
```

Comment:
the `NewMessageDispatcher` was internally invoking   `KnativeErrorTransformers` for "error decorations".

Is that part of the new logic ?

Reply:
In the new logic, we are calling it too:

https://github.com/knative/eventing/blob/64c97f15b5c755562e9a9a837f43937453473f0d/pkg/kncloudevents/event_dispatcher.go#L406

Reply:
Perhpas some extra (debug) logs would help to see what's going on?

Reply:
Found the issue in dba0268d763918833d36018611e230c7d5950e15: In https://github.com/knative/eventing/blob/e11f71adea64ec1d0bbaab17e710fff1ecba0e55/pkg/broker/filter/filter_handler.go#L257 the error was not handled before and returned a `http: wrote more than the declared Content-Length`, because the headers were not filtered and contained a `Content-Length` header.